### PR TITLE
Add pull command

### PR DESCRIPTION
**Pull request by Itxaka**
Adds the pull command to hypper

depends on: #118 

Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

File: docs/user/howto/pull.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Pulling charts to the local machine
+
+Sure, installing charts from an url or repo is cool, but sometimes you need to modify those a bit locally or just have a look at what the chart is actually doing.
+
+For that you can use `hypper pull` and that will download the chart to your local machine
```
Comment (viccuad) [inline docs/user/howto/pull.md:5]:
```suggestion
For that you can use `hypper pull` and that will download the chart to your local machine.
```

File: docs/user/howto/pull.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+```shell
+$ hypper pull hypper/fleet
+$ ls fleet*
+fleet-0.3.500.tgz
+```
+
+## Pulling a chart and extracting it
+
+Usually pull will download the tar.gz chart package, but you can use `--untar` so instead the package is extracted automatically
```
Comment (viccuad) [inline docs/user/howto/pull.md:18]:
```suggestion
Usually pull will download the tar.gz chart package, but you can use `--untar` so instead the package is extracted automatically.
```
I would either add dots (or maybe colons) to all the phrases, or remove the dot in "Pulling a specific chart version".

**Review by viccuad — DISMISSED** [review]
Nice! damn you have advanced with the docs!

**Comment by Itxaka** [comment]
@viccuad updated the last doc commit with all the missing dots! Good catch!

Also added the new doc to the generated docs!

**Comment by viccuad** [comment]
wow, now GH says "This branch has not been deployed, No deployments" on the PR. I suppose because of the changes to docs, that haven't been merged!

**Review by viccuad — APPROVED** [review]

File: cmd/hypper/pull_test.go [hunk 3]
```diff
@@ -0,0 +1,199 @@
+// +build !windows
```
Comment (viccuad) [inline cmd/hypper/pull_test.go:1]:
```suggestion
```

You are gonna hate me.. :sweat_smile:

File: docs/user/howto/pull.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+# Pulling charts to the local machine
+
+Sure, installing charts from an url or repo is cool, but sometimes you need to modify those a bit locally or just have a look at what the chart is actually doing.
```
Comment (viccuad) [inline docs/user/howto/pull.md:3]:
```suggestion
Sure, installing charts from an url or repo is cool, but sometimes you need to modify those locally a bit, or just have a look at what the chart is actually doing.
```

(if not, to me it sounds that you need a bit of locality, not that you are going to modify a bit, locally).

File: docs/user/howto/pull.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+For that you can use `hypper pull` and that will download the chart to your local machine.
```
Comment (viccuad) [inline docs/user/howto/pull.md:5]:
```suggestion
For that you can use `hypper pull`, which will download the chart to your local machine.
```

File: docs/user/howto/pull.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+```shell
+$ hypper pull hypper/fleet
+$ ls fleet*
+fleet-0.3.500.tgz
+```
+
+## Pulling a chart and extracting it
+
+Usually pull will download the tar.gz chart package, but you can use `--untar` so instead the package is extracted automatically.
```
Comment (viccuad) [inline docs/user/howto/pull.md:18]:
```suggestion
Usually, pull will download the tar.gz chart package, but you can use `--untar` so instead the package is extracted automatically.
```

File: cmd/hypper/pull_test.go (hunk 3 above)
Comment (dragonchaser) [reply cmd/hypper/pull_test.go:1]: It would be a plus to have this being run on windows as well :)

File: docs/user/howto/pull.md (hunk 4 above)
Comment (dragonchaser) [reply docs/user/howto/pull.md:3]: Also I would like to see a more technical "lingo" here like
`- Sometimes there is need to modfify charts locally.....`

**Review by dragonchaser — CHANGES_REQUESTED** [review]

File: cmd/hypper/pull_test.go (hunk 3 above)
Comment (Itxaka) [reply cmd/hypper/pull_test.go:1]: Im trying to remember why I set this....

**Comment by mattfarina** [comment]
It looks like windows tests are failing.

File: cmd/hypper/pull_test.go (hunk 3 above)
Comment (Itxaka) [reply cmd/hypper/pull_test.go:1]: yeah this requires changes on the repotest package of helm, which goes out of scope. I cannot support windows unless there is changes upstream to make it sure it works under windows as well :)

File: docs/user/howto/pull.md (hunk 4 above)
Comment (viccuad) [reply docs/user/howto/pull.md]: i consider this fixed, with https://github.com/rancher-sandbox/hypper/pull/111/commits/83b60c5855566b9a880a2030a35bf8e8f6cb241e.

**Review by viccuad — APPROVED** [review]
Approving! It would be nice to run the tests on Windows, but at least we aren't introducing a regression there.

**Review by dragonchaser — APPROVED** [review]

**Pull request merged**